README.md

```diff
@@ -42,29 +51,79 @@ Value is X-JWT-Assertion-Original.
 
 Example
 
-```
-import byu_jwt
-original_jwt_header = byu_jwt.BYU_JWT_HEADER_ORIGINAL
+```python
+original_jwt_header = byujwt.BYU_JWT_HEADER_ORIGINAL
 ```
 
-## Testing
-
-You will need a file named ~/.byu/byu-jwt-python.yaml with the following contents. Get your info at https://api.byu.edu/store/site/pages/subscriptions.jag
-```
-client_id: <your WSO2 application's Consumer Key>
-client_secret: <your WSO2 application's Consumer Secret>
-```
+### Example Python Lambda function that makes use of caching
+```python
+import byu_jwt
 
-Make sure you have python and python3 installed on your system, then install virtualenv in whatever way you install python modules (usually `$ pip install virtualenv`).
+byujwt = byu_jwt.JWT()
 
+def handler(event, context):
+    jwt_to_decode = event['headers']['X-JWT-Assertion']
+    try:
+        jwt = byujwt.decode(jwt_to_decode)
+    except Exception:
+        print("Error validating and decoding JWT")
 ```
-$ virtualenv venv
-$ source venv/bin/activate
-$ pip install -r requirements
-$ pip3 install -r requirements
-$ python byu_jwt.py True 
-# The 'True' makes the tests run in verbose mode.  
-# Leaving it off will run the tests silently and they will only print info if any tests fail.
-$ python3 byu_jwt.py True
-$ deactivate # to get out of the virtualenv
-```
+**NOTE:** The important part is putting the line `byujwt = byu_jwt.JWT()` at a global level. This allows the object to be reused on subsequent lambda invocations for as long as the lambda is warm. This allows the caching of the well-known data and respecting the cache-control headers on the certificates only refetching those when cache-control has timed out
+
+
+### Example Decoded JWT Structure
+```json
+{
+  "iss": "https://api.byu.edu",
+  "exp": 1545425710,
+  "byu": {
+    "client": {
+      "byuId": "",
+      "claimSource": "",
+      "netId": "",
+      "personId": "",
+      "preferredFirstName": "",
+      "prefix": "",
+      "restOfName": "",
+      "sortName": "",
+      "subscriberNetId": "",
+      "suffix": "",
+      "surname": "",
+      "surnamePosition": ""
+    },
+    "resourceOwner": {
+      "byuId": "",
+      "netId": "",
+      "personId": "",
+      "preferredFirstName": "",
+      "prefix": "",
+      "restOfName": "",
+      "sortName": "",
+      "suffix": "",
+      "surname": "",
+      "surnamePosition": ""
+    },
+    "webresCheck": {
```

Comment:
Again, code changes being noted in the readme, but I think we need to get rid of this. It hides the logic that is actually going on behind the scenes and as such I feel like it makes client code harder to read. The same functionality could be implemented with having functions like `getBYUId` or something like that. This just feels really obfuscated. 

Also, the webresCheck name is absolutely non-descriptive. I know this isn't your doing, but I think it's worth changing prior to a V2.

Reply:
I ended up just removing webres check completely. maybe 2.1 or 3 we can do a data class with some helper functions.